lib/tapioca/generator.rb

```diff
@@ -522,5 +543,32 @@ def compile_dsl_rbi(constant, contents)
 
       filename
     end
+
+    sig { params(tmp_dir: Pathname).returns(T.nilable(String)) }
+    def verify_dsl_rbi(tmp_dir:)
+      existing_rbis = get_file_list(dir: config.outpath).sort
+      new_rbis = get_file_list(dir: tmp_dir).sort
+
+      return "New file(s) introduced." if existing_rbis.length != new_rbis.length
+
+      desynced_files = []
+
+      (0..existing_rbis.length - 1).each do |i|
+        desynced_files << new_rbis[i] unless FileUtils.identical?(existing_rbis[i], new_rbis[i])
+      end
+
+      unless desynced_files.empty?
+        filenames = desynced_files.map { |f| f.to_s.sub!(tmp_dir.to_s, "sorbet/rbi/dsl") }.join("\n  - ")
+
+        return "File(s) updated:\n  - #{filenames}"
+      end
+
+      nil
+    end
+
+    sig { params(dir: Pathname).returns(T::Array[Pathname]) }
+    def get_file_list(dir:)
+      Dir.glob("#{dir}/**/*.rbi").reject { |e| e =~ /gems|shims/ }.map { |f| Pathname.new(f) }
```

Comment:
We shouldn't need the `reject` here since the outpath should be just the `sorbet/rbi/dsl` folder for existing files and the temp folder for the new files. We don't expect any gem or shim RBI files in there, and we could potentially decide to add more RBI folders which would mean that we would need to exclude them in the future as well.

Reply:
This was because I'm using the same method to grep the existing `./sorbet/` folder as well. Can probably clean this up.